signatures/pdf_sign_external_globalsign.go

```diff
@@ -0,0 +1,539 @@
+
+	sig.Contents = core.MakeHexString(string(data))
+	return nil
+}
+
+// IsApplicable returns true if the signature handler is applicable for the PdfSignature.
+func (es *externalSigner) IsApplicable(sig *model.PdfSignature) bool {
+	if sig == nil || sig.Filter == nil || sig.SubFilter == nil {
+		return false
+	}
+
+	return (*sig.Filter == "Adobe.PPKMS" || *sig.Filter == "Adobe.PPKLite") && *sig.SubFilter == "adbe.pkcs7.detached"
+}
+
+// NewDigest creates a new digest.
+func (es *externalSigner) NewDigest(sig *model.PdfSignature) (model.Hasher, error) {
+	return bytes.NewBuffer(nil), nil
+}
+
+// Validate validates PdfSignature.
+func (es *externalSigner) Validate(sig *model.PdfSignature, digest model.Hasher) (model.SignatureValidationResult, error) {
+	return model.SignatureValidationResult{
+		IsSigned:   true,
+		IsVerified: true,
+	}, nil
+}
+
+// getCertificateChain returns certificate chain.
+func (es *externalSigner) getCertificateChain() []*x509.Certificate {
+	return es.certChain
+}
+
+// RevocationInfoArchival is OIDAttributeAdobeRevocation attribute.
+type RevocationInfoArchival struct {
+	Crl          []asn1.RawValue `asn1:"explicit,tag:0,optional"`
+	Ocsp         []asn1.RawValue `asn1:"explicit,tag:1,optional"`
+	OtherRevInfo []asn1.RawValue `asn1:"explicit,tag:2,optional"`
+}
+
+// SignerCallback callback function of `Signer`.
+type SignerCallback func(rand io.Reader, digest []byte, opts crypto.SignerOpts) ([]byte, error)
+
+// Signer implements custom crypto.Signer which utilize globalsign DSS API
+// to sign signature digest.
+type Signer struct {
+	// sign callback
+	callback SignerCallback
+}
+
+// EncryptionAlgorithmOID returns asn1.ObjectIdentifier with value of pcks7.OIDEncryptionAlgorithmRSA.
+func (s *Signer) EncryptionAlgorithmOID() asn1.ObjectIdentifier {
+	return pkcs7.OIDEncryptionAlgorithmRSA
+}
+
+// Public returns PublicKey.
+func (s *Signer) Public() crypto.PublicKey {
+	return nil
+}
+
+// Sign with custom crypto.Signer which utilize globalsign DSS API.
+func (s *Signer) Sign(rand io.Reader, digest []byte, opts crypto.SignerOpts) (signature []byte, err error) {
+	if s.callback == nil {
+		return nil, errors.New("signer func not implemented")
+	}
+	return s.callback(rand, digest, opts)
+}
+
+// NewSigner create crypto.Signer implementation
+func NewSigner(cb SignerCallback) crypto.Signer {
+	return &Signer{
+		callback: cb,
+	}
+}
+
+// SignOption contains both digital signing
+// and annotation properties.
+type SignOption struct {
+	SignedBy string
+	Fullname string
+	Reason   string
+	Location string
+
+	// Annonate signature.
+	Annotate bool
+
+	// Position of annotation.
+	Position []float64
+
+	// Annotation font size.
+	FontSize int
+
+	// Extra signature annotation fields.
+	Extra map[string]string
+
+	FilePath string
+
+	// Just in case source file is protected
+	// and defalt password is not empty.
+	Password string
+}
+
+// createSignatureField returns signature field, signature, returns error if fails.
```

Comment:
Please rephrase this comment a bit. Perhaps something like:
```
// createSignatureField creates a signature field and an associated signature, 
// based on the specified options.
```

Reply:
Updated at 15bb7144b66b8891f5c68bd64803bb2d9ee32f54